src/main/java/online/book/store/service/imp/OrderServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderResponseDto createOrder(OrderCreateRequestDto createRequestDto, User user) {
+        Long userId = user.getId();
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findByUserId(userId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                CANT_FIND_SHOPPING_CART_BY_ID_MSG + userId));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        BigDecimal total = cartItems.stream()
+                .map(cartItem -> cartItem.getBook().getPrice().multiply(
+                        BigDecimal.valueOf(cartItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        Order order = new Order();
+        order.setShippingAddress(createRequestDto.getShippingAddress());
+        order.setUser(user);
+        order.setTotal(total);
+        orderRepository.save(order);
+
+        List<OrderItem> orderItems = cartItems.stream()
+                .map(orderItemMapper::cartItemToOrderItemWithoutOrder)
+                .peek(orderItem -> orderItem.setOrder(order))
+                .collect(Collectors.toList());
+        orderItemRepository.saveAll(orderItems);
+        order.setOrderItems(new HashSet<>(orderItems));
+
+        cartItemRepository.deleteAll(cartItems);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public OrderResponseDto setStatus(OrderStatusRequestDto statusRequestDto, Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        CANT_FIND_ORDER_BY_ID_MSG + orderId));
+        order.setStatus(statusRequestDto.getStatus());
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public Set<OrderResponseDto> getAll(Long userId) {
+        Set<Order> orders = orderRepository.findAllByUserId(userId);
+        return orders.stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItem(Long orderId, Long itemId) {
+        OrderItem orderItem = orderItemRepository
+                .findById(itemId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(CANT_FIND_ITEM_BY_ID_MSG + itemId));
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Override
+    public Set<OrderItemResponseDto> getAllOrderItems(Long orderId) {
+        List<OrderItem> orderItems = orderItemRepository.findAllByOrderId(orderId);
+        return orderItems.stream()
+                .map(orderItemMapper::toDto)
+                .collect(Collectors.toSet());
```

Comment:
```suggestion
        return orderItemMapper.toDtoList(orderItems);
```
try to use method annotated by `@IterableMapping`